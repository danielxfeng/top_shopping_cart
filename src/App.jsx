import { Outlet } from "react-router-dom";
import { useState, useEffect } from "react";
import Header from "./components/Header";
import Footer from "./components/Footer";
import productsApi from "./services/productsApi";

const App = () => {
  const [productApiLoaded, setProductApiLoaded] = useState(false);

  useEffect(() => {
    const isLoaded = async () => {
      try {
        await productsApi.init();
        setProductApiLoaded(true);
      } catch (error) {
        console.error(error);
      }
    }

    isLoaded();
  }, []);

  if (!productApiLoaded) return <div>Loading...</div>;

  return (
    <>
      <Header />
      <main>
        <Outlet />
      </main>
      <Footer />
    </>
  );
};

export default App;
