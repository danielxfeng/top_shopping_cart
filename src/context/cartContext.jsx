import { createContext, useState, useContext } from 'react';

const CartContext = createContext();

// CartProvider is a component that provides the cart context to its children.
const CartProvider = ({ children }) => {
  const [updateSign, setUpdateSign] = useState({});

  return (
    <CartContext.Provider value={{ updateSign, setUpdateSign }}>
      {children}
    </CartContext.Provider>
  );
}

// The custom hook for easily accessing the context.
const useCartUpdateSign = () => {
  const context = useContext(CartContext);
  if (context === undefined) {
    throw new Error('useCart must be used within a CartProvider');
  }
  return context;
}

export { CartProvider, useCartUpdateSign };
