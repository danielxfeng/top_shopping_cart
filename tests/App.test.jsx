import { describe, it, expect } from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import { CartProvider } from "../src/context/cartContext";
import { MemoryRouter } from "react-router-dom";
import App from "../src/App";

describe("App page", () => {
  it("renders correctly", async () => {
    const contextValue = { updateSign: {} };
    const { container } = render(
      <MemoryRouter>
        <CartProvider value={contextValue}>
          <App />
        </CartProvider>
      </MemoryRouter>
    );
    await waitFor(() => screen.getByText(/Home/i));
    expect(container).toMatchSnapshot();
  });
});
