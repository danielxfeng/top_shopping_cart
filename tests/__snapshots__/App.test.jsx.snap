// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App page > renders correctly 1`] = `
<div>
  <header
    class="_header_a9d33c"
  >
    <div
      class="_left_a9d33c"
    >
      <a
        class="_title_a9d33c"
        href="/"
      >
        <p>
          Fancy Mall
        </p>
      </a>
      <nav
        class="_nav_a9d33c"
      >
        <ul>
          <li>
            <a
              class="_nav__primary_a9d33c"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <div
              class="_nav__dropdown_a9d33c"
            >
              <a
                class="_nav__primary_a9d33c"
                href="/store"
              >
                Store
                <span
                  class="_drop__down__sign_a9d33c"
                >
                  ⌵
                </span>
              </a>
              <div
                class="_nav__secondary_a9d33c"
              >
                <ul
                  class="_nav__secondary__ul_a9d33c"
                >
                  <li>
                    <a
                      href="/store/electronics"
                    >
                      electronics
                    </a>
                  </li>
                  <li>
                    <a
                      href="/store/jewelery"
                    >
                      jewelery
                    </a>
                  </li>
                  <li>
                    <a
                      href="/store/men's clothing"
                    >
                      men's clothing
                    </a>
                  </li>
                  <li>
                    <a
                      href="/store/women's clothing"
                    >
                      women's clothing
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="_cart_a9d33c"
    >
      <a
        href="/cart"
      >
        <img
          alt="shopping cart"
          class="_cart__png_a9d33c"
          src="/src/assets/shopping_cart.png"
        />
      </a>
    </div>
  </header>
  <main
    class="_main_61351a"
  />
  <footer
    class="_footer_4d8704"
  >
    <p>
      <span>
        © 2024
      </span>
      <a
        href="https://github.com/danielxfeng/top_shopping_cart"
        rel="noopener noreferrer"
        target="_blank"
      >
        Fancy Mall
      </a>
    </p>
    <p>
      Made by ❤️
      <em>
        Daniel
      </em>
      ❤️
    </p>
  </footer>
</div>
`;
